src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -488,65 +510,75 @@ public void createSnapshotWithLiveViews(
         Snapshot snapshot,
         Dataset dataset) throws InterruptedException {
 
-        BigQueryProject bigQueryProject = bigQueryProjectForSnapshot(snapshot);
-        String projectId = bigQueryProject.getProjectId();
+        BigQueryProject snapshotBigQueryProject = bigQueryProjectForSnapshot(snapshot);
+        String snapshotProjectId = snapshotBigQueryProject.getProjectId();
         String snapshotName = snapshot.getName();
-        BigQuery bigQuery = bigQueryProject.getBigQuery();
+        BigQuery bigQuery = snapshotBigQueryProject.getBigQuery();
 
         String datasetBqDatasetName = prefixName(dataset.getName());
+        BigQueryProject datasetBigQueryProject = bigQueryProjectForDataset(dataset);
+        String datasetProjectId = datasetBigQueryProject.getProjectId();
 
         // create snapshot BQ dataset
-        snapshotCreateBQDataset(bigQueryProject, snapshot);
+        snapshotCreateBQDataset(snapshotBigQueryProject, snapshot);
 
         // create the row id table (row id col and table id col)
-        bigQueryProject.createTable(snapshotName, PDAO_ROW_ID_TABLE, rowIdTableSchema());
+        snapshotBigQueryProject.createTable(snapshotName, PDAO_ROW_ID_TABLE, rowIdTableSchema());
 
         // get source dataset table live views
         List<DatasetTable> tables = dataset.getTables();
 
         // create a snapshot table based on the live view data row ids
-        String liveViewTables = createSnaphotTableFromLiveViews(bigQueryProject, tables, datasetBqDatasetName);
+        //TODO - the big query project is used to reference the dataset here
+        String liveViewTables = createSnaphotTableFromLiveViews(snapshotBigQueryProject, tables, datasetBqDatasetName);
```

Comment:
Right: I think we might need to pass in `snapshotBigQueryProject` and `datasetBigQueryProject`